command/agent/command.go

```diff
@@ -372,6 +372,12 @@ func (c *Command) isValidConfig(config, cmdConfig *Config) bool {
 		return false
 	}
 
+	if config.Client.MinDynamicPort > 0 && config.Client.MaxDynamicPort > 0 &&
+		config.Client.MinDynamicPort >= config.Client.MaxDynamicPort {
+		c.Ui.Error("Invalid dynamic port range")
+		return false
```

Comment:
I think we need additional validation here to ensure both values are within acceptable ranges. In local testing I was able to set negative values, which are then ignored within `nomad/structs` because the `SetNode` check looks for values greater than zero.